include/sympack/symPACKMatrix2D.hpp

```diff
@@ -651,6 +726,7 @@ namespace symPACK{
         }
 
         // Copy constructor.  
+        // TODO: Edit this with new fields
```

Comment:
IIUC the `blockCell_t` copy/move constructors and copy/move assignment have not been updated to handle the new CUDA_MODE fields (`_d_nzval`, `is_gpu_block`). I'd expect this to cause potentially subtle problems **if** these are ever invoked in CUDA_MODE, and the short comment above is unlikely to help find the problem.

Are these operations simply unreachable?

If so, then for maintainability reasons I recommend inserting the following in each such function:
```
#ifdef CUDA_MODE
throw std::runtime_error("This function is unimplemented for CUDA_MODE").
#endif
```

Reply:
These constructors/operators are only called during the Solve phase, which does not currently use the strategy that the Factorize phase uses to bypass the intermediate copy to host memory in certain cases. As a result, it is impossible at any point for a `blockCell_t` instance that these constructors/operators interact with to have either the `_d_nzval` or `is_gpu_block` field set to anything other than `nullptr` and `false`, respectively. 

I will add a more detailed comment to the code explaining that these constructors will need to be edited if someone ever wants to add the capability to bypass host memory to the communication present in the Solve phase.

Reply:
> it is impossible at any point for a `blockCell_t` instance that these constructors/operators interact with to have either the `_d_nzval` or `is_gpu_block` field set to anything other than `nullptr` and `false`, respectively.
> 
> I will add a more detailed comment to the code explaining that these constructors will need to be edited if someone ever wants to add the capability to bypass host memory to the communication present in the Solve phase.

Thanks for the clarification.

You've stated a behavioral invariant, and IMO such invariants are always much more useful when embodied as an assertion rather than a comment.

How about inserting the following in each function:
```
#ifdef CUDA_MODE
UPCXX_ASSERT(!_d_nzval && !is_gpu_block, "This function is unimplemented for blocks using host-bypass communication");
#endif
```

Reply:
Fixed here [d7d69d0](https://github.com/symPACK/symPACK/pull/12/commits/d7d69d0a4241a680d3940fc9ea807395f75a0548)